deploy/lib/xquery/triggers.xqy

```diff
@@ -0,0 +1,97 @@
+(:
+Copyright 2016 MarkLogic Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+:)
+xquery version "1.0-ml";
+
+import module namespace trgr="http://marklogic.com/xdmp/triggers" at "/MarkLogic/triggers.xqy";
+
+declare namespace triggers = "http://marklogic.com/roxy/triggers";
+
+declare option xdmp:mapping "false";
+
+(:
+ : Loads a pipeline from a configuration xml
+ :
+ :@param $config - the configuration xml
+ :
+:)
+declare function triggers:load-from-config($config as element(trgr:triggers))
+{
+  let $triggers-db := xdmp:database()
+  for $trgr in $config/trgr:trigger
+  let $name := $trgr/trgr:name
+  let $desc as xs:string := $trgr/trgr:description
+  let $data-event := $trgr/trgr:data-event
+  let $module :=
+    <trgr:module>
+      <trgr:database>{xdmp:database($trgr/trgr:module/trgr:database/fn:string())}</trgr:database>
+      {
+        $trgr/trgr:module/* except $trgr/trgr:module/trgr:database
+      }
+    </trgr:module>
```

Comment:
I mention recrearing the xml with built-in functions because humans can put in things that cause trouble down the line (we have had that with indexes), and the built-in xml format could change (like addition of new elements). Not highly likely does make it more robust i think..